### PR TITLE
API add stops details to the report

### LINKED ISSUE #1911: Add track number to the simulation report
## Description

Add track number to the payload of the report response

## Goal

Being able to build a train driver schedule

## Acceptance criteria

- [x] The track number must be in the endpoint response

## Definition of ready

- [ ] The enabler story is well defined
- [ ] Acceptance criterias are defined
- [ ] The individual responsible for validating the enbaler story is assigned
- [ ] Does not depend on an missing feature
- [ ] Negociable with the implementation team
- [ ] Valuable to the end users
- [ ] Estimable by the implementation team
- [ ] As small as it can be
- [ ] Testable by the implementation team

## Definition of done

- [x] Acceptance criteria are met
- [x] Automated tests are written against acceptance criteria
- [x] Story ok-ed by Product Owner
- [x] Documentation updated

### PR DESCRIPTION
**Pull request by flomonster**
close #1911

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/TrackSectionImpl.java [hunk 1]
```diff
@@ -20,6 +20,8 @@ public class TrackSectionImpl implements TrackSection {
     private final LineString geo;
     private final LineString sch;
     private final ImmutableRangeMap<Double, LoadingGaugeConstraint> loadingGaugeConstraints;
+    private int trackNumber = 0;
+    private int lineCode = 0;
```
Comment (eckter) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/TrackSectionImpl.java]: These can be final

File: tests/fuzzer/fuzzer.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ class FailedTest(Exception):
 
 
 class ErrorType(str, Enum):
-    PATHFINDING = "PATHFINDING",
+    PATHFINDING = ("PATHFINDING",)
```
Comment (eckter) [inline tests/fuzzer/fuzzer.py]: I'm not really sure what this changes, but it bothers me a bit that it's not consistent anymore, can it be changed for the other values as well?

File: core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/TrackSectionImpl.java (hunk 1 above)
Comment (flomonster) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/TrackSectionImpl.java]: It cannot since it's used in the constructor

Comment (multun) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/TrackSectionImpl.java]: try removing the default value of zero and adding final

**Review by multun — APPROVED** [review]

**Pull request merged**